docs/user-guide/bestpractices.rst

```diff
@@ -0,0 +1,19 @@
+.. _data-curator-best-practices:
+
+======================================
+Best Practices
+======================================
+
+------------------------------
+Choosing the Right Model Size
+------------------------------
+NeMo Curator offers a variety of methods for determining the quality of a piece of text.
+Here are the methods in increasing order of compute required for them.
+
+#. `fastText <https://docs.nvidia.com/nemo-framework/user-guide/latest/datacuration/qualityfiltering.html#classifier-filtering>`_ is an n-gram based bag-of-words classifier. It is typically trained on a high quality reference corpus and a low quality corpus (typically unfiltered Common Crawl dumps). While NeMo Curator does not provide pretrained versions of the classifier, training it is incredibly fast and easy. It only requires 100,000 - 1,000,000 text samples to train on, and can complete training in mere seconds. Its small size also allows it to train and run inference on the CPU. Due to these factors, we recommend using fastText classifiers on large scale pretraining datasets where you don't have the compute budget for more sophisticated methods.
+
+#. `BERT-style classifiers <https://docs.nvidia.com/nemo-framework/user-guide/latest/datacuration/distributeddataclassification.html>`_ - NeMo Curator's distributed data classification modules work with many BERT-style classifiers for domain classification, quality classification, and more. For this comparison, we'll focus on just the text quality classifier. NeMo Curator provides a pretrained version of the classifier on HuggingFace and NGC that can be immediately used. We recommend using these classifiers towards the end of your data filtering pipeline for pretraining.
+
+#. `Language model labelling <https://docs.nvidia.com/nemo-framework/user-guide/latest/datacuration/syntheticdata.html>`_ - Language models can be used to label text has high quality or low quality. NeMo Curator allows you to connect to arbitrary LLM inference endpoints which you can use to label your data. However, due to their size, these models can require a lot of compute and are usually infeasible to run across an entire pretraining dataset. We recommend using these large models on very little amounts of data. Fine-tuning datasets could make good use of them. If you still want to leverage an LLM's output on a larger scale, one alternative is to label a small subset of your data and train a smaller BERT-style classifier on the labeled data. This classifier can then be plugged-in to our distributed data classification module and applied to a larger dataset.
```

Comment:
Link is broken for me.

"one alternative is to label a small subset of your data and train a smaller BERT-style classifier on the labeled data" - to clarify, is this supposed to refer to the above bullet point? If so, it might be helpful to add something like "or use our BERT-style quality classifier (link to HuggingFace here)".

Reply:
> Link is broken for me.

Yep, the link should be correct though when we release it on the user guide.

> to clarify, is this supposed to refer to the above bullet point?

Sort of. The intention is to say that they can train a bert model and use it with distributed data classification module by writing their own version of our quality classifier. How can I make that clearer?

Reply:
Ok, is there an example somewhere for how a user could do this? I think it could also be helpful to link to an example of our distributed data classification module too in this case.

Reply:
Hm there isn't a good example that demonstrates how a user can plug in their own model into the distributed data classification pipeline. I'd like to link to the translation example but that's not merged in. Do you think I should just remove this line in the meantime?

Reply:
Yeah I think it might be better to remove for now, but in that case is there an example model we can link to in the meantime? Mainly asking because we are able to give concrete examples for all the other model types in this guide.

Also, it looks like there's a small typo - it should be "Language models can be used to label text AS high quality or low quality", not has.

Reply:
Good callouts. I added a reference to Nemotron-4 340B Instruct on build.nvidia.com. Let me know what you think now.

Reply:
LGTM!